src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediatePreparedStatement.java

```diff
@@ -78,34 +56,25 @@ private int parseType(byte[][] parameters, int index) throws IllegalArgumentExce
     if (this.parameterDataTypes.size() > index) {
       return this.parameterDataTypes.get(index);
     } else {
-      throw new IllegalArgumentException("PgAdapter does not support untyped parameters.");
+      return Oid.UNSPECIFIED;
```

Comment:
We must support untyped parameters. At least the JDBC driver will send some parameters untyped (notably all date/time parameters).